ldap_sync/sources/db.py

```diff
@@ -0,0 +1,261 @@
+        ).all(),
+    )
+
+
+def fetch_db_groups(
+    session: Session,
+    base_dn: types.DN,
+    user_base_dn: types.DN,
+) -> typing.Iterator[GroupRecord]:
+    """Fetch the groups to be synced (in the form of :class:`GroupRecords <GroupRecord>`).
+
+    :param session: the SQLAlchemy database session
+    :param base_dn: the group base dn
+    :param user_base_dn: the base dn of users. Used to infer DNs of the group's members.
+    """
+    for res in _fetch_db_groups(session):
+        yield conversion.db_group_to_record(
+            name=res.Group.name,
+            members=res.members,
+            base_dn=base_dn,
+            user_base_dn=user_base_dn,
+        )
+
+
+class _PropertyProxyType(NamedTuple):
+    """Representation of a property as returned by :func:`_fetch_db_properties`."""
+
+    name: str
+    members: list[str]
+
+
+def _fetch_db_properties(session: Session) -> list[_PropertyProxyType]:
+    """Fetch the groups who should be synced.
+
+    Explicitly, this returns everything in :ref:`EXPORTED_PROPERTIES` together with
+    the current users having the respective property as members.
+
+    :param session: The SQLAlchemy session to use
+
+    :returns: An iterable of `(property_name, members)` ResultProxies.
+    """
+    properties = session.execute(
+        select(
+            CurrentProperty.property_name.label("name"),
+            func.array_agg(User.login).label("members"),
+        )
+        .select_from(
+            join(CurrentProperty, User, onclause=CurrentProperty.user_id == User.id)
+        )
+        .where(CurrentProperty.property_name.in_(EXPORTED_PROPERTIES))
+        .group_by(CurrentProperty.property_name)
+    ).fetchall()
+
+    missing_properties = EXPORTED_PROPERTIES - {p.name for p in properties}
+    # Return mutable copy instead of SQLAlchemy's immutable RowProxy
+    return [_PropertyProxyType(p.name, p.members) for p in properties] + [
+        _PropertyProxyType(p, []) for p in missing_properties
+    ]
+
+
+def fetch_db_properties(
+    session: Session,
+    base_dn: types.DN,
+    user_base_dn: types.DN,
+) -> typing.Iterator[GroupRecord]:
+    """Fetch the properties to be synced (in the form of :class:`GroupRecords <GroupRecord>`).
+
+    :param session: the SQLAlchemy database session
+    :param base_dn: the property base dn
+    :param user_base_dn: the base dn of users. Used to infer DNs of the users who are currently
+        carrying this property.
+    """
+    for res in _fetch_db_properties(session):
+        yield conversion.db_group_to_record(
+            name=res.name,
+            members=res.members,
+            base_dn=base_dn,
+            user_base_dn=user_base_dn,
+        )
+
+
+EXPORTED_PROPERTIES = frozenset(
+    [
+        "network_access",
+        "mail",
+        "traffic_limit_exceeded",
+        "payment_in_default",
+        "violation",
+        "member",
+        "ldap_login_enabled",
+        "userdb",
+        "cache_access",
```

Comment:
Can be removed

Reply:
Wait, why? This field is used in line 220, and logically, we have to restrict which properties we export

Reply:
Oh, you meant `cache_access`. Alright.
```suggestion
```